testing/cli/test_cli_sct_maths.py

```diff
@@ -1,18 +1,57 @@
 # pytest unit tests for sct_maths
 
+import os
 import pytest
 import traceback as tb
 import logging
 
 import numpy as np
 
-from spinalcordtoolbox.image import Image
+from spinalcordtoolbox.image import Image, compute_dice
 from spinalcordtoolbox.utils.sys import sct_test_path
 from spinalcordtoolbox.scripts import sct_maths
 
 logger = logging.getLogger(__name__)
 
 
+@pytest.mark.parametrize("ofolder", ["tmp_path", None])
+@pytest.mark.parametrize("args_to_test, fname_in, fname_gt", [
+    (["-denoise", "1"],       sct_test_path("t2", "t2.nii.gz"), None),  # TODO: Add gt
+    (["-otsu-median", "5,5"], sct_test_path("t2", "t2.nii.gz"), None),  # TODO: Add gt
+], ids=["-denoise", "-otsu-median"])
+def test_sct_maths_basic(args_to_test, fname_in, fname_gt, ofolder, tmp_path):
+    """Basic test to ensure core functionality runs without error."""
+    # base args to test
+    args = ["-i", fname_in]
+    args.extend(args_to_test)
+
+    # test all combinations of both (user-supplied and default) output filenames/folders
+    fname_out = "img_out.nii.gz"
+    path_out = (str(tmp_path) if ofolder == "tmp_path" else os.getcwd())
+    args.extend(["-o", os.path.join(path_out, fname_out) if ofolder else fname_out])
+
+    # run script
+    sct_maths.main(argv=args)
+
+    # test all output files exist
+    outfiles = [fname_out]
```

Comment:
I get that the template needs to account for multiple output files, but maybe when there's a single one we could inline `outfiles` and unroll the resulting single-iteration loops? Possibly also give a nice name to `os.path.join(path_out, fname_out)` which gets repeated 4 times?

(I'm saying this because this hardcoded value is part of the test body itself, rather than one of the parameters. By contrast, I think it's fine to keep the `if fname_gt` bit, even though `fname_gt` is currently always `None`, because that's a more contingent fact.)

Reply:
Absolutely!